docs/visualization/rgb.rst

```diff
@@ -91,3 +91,157 @@ of the galaxies show up. Compare with Fig. 1 of `Lupton et al. (2004)`_ or the
+
+    # Use the maximum value of the 99.95% percentile over all three filters
+    # as the maximum value:
+    pctl = 99.95
+    maximum = 0.
+    for img in [i,r,g]:
+        val = np.percentile(img,pctl)
+        if val > maximum:
+            maximum = val
+
+    rgb_log = make_log_rgb(i, r, g, minimum=0., maximum=maximum, scalea=1000,
+                           filename="ngc6976-log.jpeg")
+    plt.imshow(rgb_log, origin='lower')
+
+By specifying per-filter maximum values, it is possible to emphasize
+certain objects, such as the very reddest sources:
+
+.. plot::
+   :context:
+   :include-source:
+   :align: center
+
+    # Increase the red maximum to emphasize the very reddest sources:
+    maximum = 3 * [maximum]
+    maximum[0] = 30.
+    rgb_log = make_log_rgb(i, r, g, minimum=0, maximum=maximum, scalea=1000,
+                           filename="ngc6976-log-alt.jpeg")
+    plt.imshow(rgb_log, origin='lower')
+
+
+Finally, we construct an example linear scaling RGB image using the
+same normalization bounds as the first logarithmic scaling example.
+
+.. plot::
+   :context:
+   :include-source:
+   :align: center
+
+    from astropy.visualization import make_linear_rgb
+
+    # Use the maximum value of the 99.5% percentile over all three filters
+    # as the maximum value:
+    pctl = 99.5
+    maximum = 0.
+    for img in [i,r,g]:
+        val = np.percentile(img,pctl)
+        if val > maximum:
+            maximum = val
+    rgb_log = make_linear_rgb(i, r, g, minimum=0, maximum=maximum,
+                              filename="ngc6976-linear.jpeg")
+    plt.imshow(rgb_log, origin='lower')
+
+
+Here the linear stretch greatly restricts the visible dynamic range,
+but for data with limited dynamic range this scaling will better
+emphasize features than logarithmic scaling.
+
+
+.. _astropy-visualization-rgb-user-stretch:
+
+RGB images using user-specified stretch
+=======================================
+
+
+It is also possible to specify a different scaling to be applied to all
+three RGB images separately using the convenience function
+:func:`~astropy.visualization.make_rgb`, which takes an instance of
+a subclass of :class:`~astropy.visualization.BaseStretch` in addition to
+``minimum`` and ``maximum`` to specify the normalization.
+
+.. plot::
+   :context: reset
+   :include-source:
+   :align: center
+
+    import numpy as np
+    import matplotlib.pyplot as plt
+    from astropy.visualization import make_rgb, SqrtStretch
+    from astropy.io import fits
+    from astropy.utils.data import get_pkg_data_filename
+
+    # Read in the three images downloaded from here:
+    g_name = get_pkg_data_filename('visualization/reprojected_sdss_g.fits.bz2')
+    r_name = get_pkg_data_filename('visualization/reprojected_sdss_r.fits.bz2')
+    i_name = get_pkg_data_filename('visualization/reprojected_sdss_i.fits.bz2')
+    g = fits.open(g_name)[0].data
+    r = fits.open(r_name)[0].data
+    i = fits.open(i_name)[0].data
```

Comment:
Not sure how long it takes to generate one file, but maybe we should limit the number of examples, our documentation is already quite long to build.